conan/internal/api/detect/detect_api.py

```diff
@@ -347,6 +347,50 @@ def detect_cppstd(compiler, compiler_version):
     return cppstd
 
 
+def default_cstd(compiler, compiler_version):
+    """returns the default cstd for the compiler-version. This is not detected, just the default"""
+
+    def _clang_cstd_default(version):
+        if version >= "11":
+            return "gnu17"  # https://releases.llvm.org/11.0.0/tools/clang/docs/ReleaseNotes.html#c-language-changes-in-clang
+        elif version >= "4":  # 3.5 actually
+            return "gnu11"
+        else:
+            return "gnu99"  # It was gnu89 actually
+
+    def _gcc_cstd_default(version):
+        if version >= "15":  # https://www.gnu.org/software/gcc/gcc-15/changes.html#c
+            return "gnu23"
+        elif version >= "8":
+            return "gnu17"  # https://www.gnu.org/software/gcc/gcc-8/changes.html#c
+        elif version >= "5":
+            return "gnu11"  # https://www.gnu.org/software/gcc/gcc-5/changes.html#c
+        else:
+            return "gnu99"  # It was gnu89 actually
+
+    def _visual_cstd_default(version):
+        return None
+
+    def _apple_clang_cstd_default(version):
+        return None
```

Comment:
```suggestion
        # Based on which LLVM/Clang versions these are based on
        if version >= "12":
            return "gnu17"
        if version >= "10":
            return "gnu11"
        return "gnu99"
```

Reply:
Sorry, didn't have time yesterday to do it. Thank you for your help!